From fba93b250c0d566f7ef26442312286310b2b9b46 Mon Sep 17 00:00:00 2001
From: Father Chrysostomos <sprout@cpan.org>
Date: Sun, 2 Jun 2013 00:36:33 -0700
Subject: [PATCH] =?UTF-8?q?[perl=20#118237]=20Fix=20coreamp.t=E2=80=99s=20?=
 =?UTF-8?q?rand=20test?=
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

when rand returns something really small that does not
begin with 0, such as 2.90736361456823e-05.
---
 t/op/coreamp.t | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/t/op/coreamp.t b/t/op/coreamp.t
index c1f7181..fe7c741 100644
--- a/t/op/coreamp.t
+++ b/t/op/coreamp.t
@@ -637,7 +637,7 @@ test_proto 'quotemeta', '$', '\$';
 
 test_proto 'rand';
 $tests += 3;
-like &CORE::rand, qr/^0[.\d+-e]*\z/, '&rand';
+like &CORE::rand, qr/^[.\d+-e]*\z/, '&rand';
 unlike join(" ", &CORE::rand), qr/ /, '&rand in list context';
 &cmp_ok(&CORE::rand(78), qw '< 78', '&rand with 1 arg');
 
-- 
1.8.3.1

